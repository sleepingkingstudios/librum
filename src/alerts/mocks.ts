import type {
  Alert,
  DismissAlert,
  DismissAllAlerts,
  DisplayAlert,
} from '@alerts';

type MockAlertsContext = {
  alerts: Alert[],
  dismissAlert: jest.MockedFunction<DismissAlert>,
  dismissAllAlerts: jest.MockedFunction<DismissAllAlerts>,
  displayAlert: jest.MockedFunction<DisplayAlert>,
};

type MockUseAlerts = jest.MockedFunction<() => MockAlertsContext>;

const alerts: MockAlertsContext = {
  alerts: [] as Alert[],
  dismissAlert: jest.fn(),
  dismissAllAlerts: jest.fn(),
  displayAlert: jest.fn(),
};

export const useAlerts: MockUseAlerts = jest.fn(() => alerts);
