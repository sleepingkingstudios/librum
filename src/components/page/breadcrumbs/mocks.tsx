import * as React from 'react';

import type { Breadcrumb } from './types';

const formatBreadcrumb = (breadcrumb: Breadcrumb): string => {
  const { active, label, url } = breadcrumb;

  return `${label}${url ? ` @ ${url}` : ''}${active ? ' (active)' : ''}`;
};

export const PageBreadcrumbs = ({
  breadcrumbs,
}: {
  breadcrumbs: Breadcrumb[],
}): JSX.Element => (
  <ul aria-label="breadcrumbs">
    {
      breadcrumbs.map((breadcrumb: Breadcrumb) => (
        <li aria-label="breadcrumb" key={breadcrumb.label}>
          { formatBreadcrumb(breadcrumb) }
        </li>
      ))
    }
  </ul>
);
