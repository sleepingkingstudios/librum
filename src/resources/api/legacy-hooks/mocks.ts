import { defaultResponse } from '@api/test-helpers';
import type { UseResourceQuery } from '../types';

const resourceQuery: jest.MockedFunction<UseResourceQuery> =
  jest.fn(() => defaultResponse);

export const useResourceQuery: jest.MockedFunction<() => UseResourceQuery> =
  jest.fn(() => resourceQuery);
