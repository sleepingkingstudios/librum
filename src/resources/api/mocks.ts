import type { UseQuery } from '@api';
import { defaultResult } from '@api/test-helpers';
import type {
  ResourceApiEndpointConfiguration,
  ResourceConfiguration,
} from '../types';
import type { ResourceApiHooks } from './types';

const useIndexResources: jest.MockedFunction<UseQuery> =
  jest.fn(() => defaultResult);

type GenerateResourcesApiProps = {
  endpoints?: ResourceApiEndpointConfiguration,
} & ResourceConfiguration;

type GenerateResourcesApi =
  (options: GenerateResourcesApiProps) => ResourceApiHooks;

export const generateResourcesApi: jest.MockedFunction<GenerateResourcesApi> =
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  jest.fn((...args) => ({
    useIndexResources,
  }));
