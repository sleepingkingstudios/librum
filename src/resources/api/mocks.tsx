import { responseWithStatus } from '@api';
import type {
  Refetch,
  Response,
} from '@api';

type UseResourceQuery = (props: unknown) => [Response, Refetch];

export { generateAlerts } from './utils';

export const useResourceQuery: jest.MockedFunction<UseResourceQuery> = jest.fn(
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  (props) => [responseWithStatus({ status: 'uninitialized' }), jest.fn()]
);
