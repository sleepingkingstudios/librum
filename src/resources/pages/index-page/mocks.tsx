import * as React from 'react';

import { responseWithData } from '@api';
import type { DataTableData } from '@components/data-table';
import { ResourcePage } from '@resources/components/page';
import type { ResourcePageProps } from '@resources/components/page';

export const MockTable = ({ data }: { data: DataTableData }): JSX.Element => {
  const books = 'rareBooks' in data ? data.rareBooks : [];

  return (
    <span className="mock-data-table">
      { `There are ${books.length} books!` }
    </span>
  );
};

export const ResourceIndexPage = ({
  Table,
  page,
  resourceName,
}: ResourcePageProps): JSX.Element => {
  const data: DataTableData = {
    rareBooks: [
      {
        title: 'On War',
      },
      {
        title: 'The Art of War',
      },
      {
        title: 'The Prince',
      },
    ],
  };
  const response = responseWithData({ data });
  const contents = (
    <div>
      <h1>Content for {resourceName}</h1>

      <Table data={data} name={resourceName} />
    </div>
  );
  const pageWithDefaults = {
    ...page,
    contents,
  };

  return (
    <ResourcePage
      action={''}
      page={pageWithDefaults}
      resourceName={resourceName}
      response={response}
    />
  );
};
