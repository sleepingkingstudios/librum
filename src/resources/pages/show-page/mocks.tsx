import * as React from 'react';

import { responseWithData } from '@api';
import { DataList } from '@components/data-list';
import { ResourcePage } from '@resources/components/page';
import type { ResourcePageProps } from '@resources/components/page';
import type { DataBlockData } from '@resources/types';
import { singularize } from '@utils/inflector';
import { Literal } from '@utils/types';

const defaultBook = {
  name: 'Unknown Book',
  slug: 'unknown-book',
  author: 'Unknown Author',
};

export const MockBlock = ({ data }: { data: DataBlockData }): JSX.Element => {
  const book = (data.rareBook || defaultBook) as Record<string, Literal>;

  return (<DataList data={book} />);
};

export const ResourceShowPage = ({
  Block,
  page,
  resourceName,
}: ResourcePageProps): JSX.Element => {
  const data: DataBlockData = {
    rareBook: {
      name: 'Gideon the Ninth',
      slug: 'gideon-9',
      author: 'Tammsyn Muir',
    },
  };
  const response = responseWithData({ data });
  const contents = (
    <div>
      <h1>Content for {singularize(resourceName)}</h1>

      <Block data={data} name={resourceName} />
    </div>
  );

  const pageWithDefaults = {
    ...page,
    contents,
  };

  return (
    <ResourcePage
      action={''}
      page={pageWithDefaults}
      resourceName={resourceName}
      response={response}
    />
  );
};
