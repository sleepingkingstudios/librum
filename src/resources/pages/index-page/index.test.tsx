import * as React from 'react';

import '@testing-library/jest-dom';

import { ResourceIndexPage } from './index';
import {
  responseWithData,
  responseWithStatus,
} from '@api';
import {
  generateAlerts,
  useResourceQuery,
} from '@resources/api';
import type { ResourcePageOptions } from '@resources/components/page';
import { ConfiguredDataTable } from '@resources/types';
import { render } from '@test-helpers/rendering';
import { MockTable as Table } from './mocks';

// eslint-disable-next-line @typescript-eslint/no-unsafe-return
jest.mock('@resources/api', () => require('@resources/api/mocks'));
// eslint-disable-next-line @typescript-eslint/no-unsafe-return
jest.mock('@components/page', () => require('@components/page/mocks'));
jest.mock(
  '@components/page/breadcrumbs',
// eslint-disable-next-line @typescript-eslint/no-unsafe-return
  () => require('@components/page/breadcrumbs/mocks'),
);

const mockUseResourceQuery = useResourceQuery as jest.MockedFunction<
  typeof useResourceQuery
>;

describe('<ResourceIndexPage />', () => {
  const action = 'index';
  const page: ResourcePageOptions = { member: false };
  const resourceName = 'rareBooks';
  const renderIndexPage = ({
    Table,
  }: {
    Table?: ConfiguredDataTable,
  }) => render(
    <ResourceIndexPage
      Table={Table}
      action={action}
      page={page}
      resourceName={resourceName}
    />,
    {
      router: true,
      store: true,
    },
  );
  const refetch = jest.fn();
  const response = responseWithStatus({ status: 'uninitialized' });

  beforeEach(() => {
    mockUseResourceQuery
      .mockClear()
      .mockImplementation(() => [response, refetch]);
  });

  it('should configure the request', () => {
    const alerts = generateAlerts({
      action: 'find',
      member: false,
      query: true,
      resourceName,
    });
    const expected = {
      action,
      alerts,
      member: false,
      resourceName,
      url: '',
    };

    renderIndexPage({});

    expect(useResourceQuery).toHaveBeenCalledWith(expected);
  });

  it('should display the breadcrumbs', () => {
    const { getAllByRole } = renderIndexPage({});
    const expected = [
      'Home @ /',
      'Rare Books (active)',
    ];

    const rendered = getAllByRole('listitem');

    expect(rendered).toHaveLength(2);
    expect(rendered.map(item => item.textContent)).toEqual(expected);
  });

  it('should display the contents', () => {
    const { getByText } = renderIndexPage({});

    const contents = getByText('Missing <Table /> Component');
    expect(contents).toBeVisible();
  });

  it('should display the heading', () => {
    const { getByRole } = renderIndexPage({});

    const heading = getByRole('heading', { level: 1 });

    expect(heading).toBeVisible();
    expect(heading).toHaveTextContent('Rare Books');
  });

  it('should match the snapshot', () => {
    const { asFragment } = renderIndexPage({});

    expect(asFragment()).toMatchSnapshot();
  });

  describe('when the query returns a failing response', () => {
    const response = responseWithStatus({ status: 'failure' });

    beforeEach(() => {
      mockUseResourceQuery.mockImplementation(() => [response, refetch]);
    });

    it('should display the contents', () => {
      const { getByText } = renderIndexPage({});

      const contents = getByText('Missing <Table /> Component');
      expect(contents).toBeVisible();
    });

    it('should match the snapshot', () => {
      const { asFragment } = renderIndexPage({});

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when the query returns a loading response', () => {
    const response = responseWithStatus({ status: 'loading' });

    beforeEach(() => {
      mockUseResourceQuery.mockImplementation(() => [response, refetch]);
    });

    it('should display the loading message', () => {
      const { getByText } = renderIndexPage({});

      const contents = getByText('Loading Rare Books...');
      expect(contents).toBeVisible();
    });

    it('should match the snapshot', () => {
      const { asFragment } = renderIndexPage({});

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('when the query returns a successful response', () => {
    const response = responseWithStatus({ status: 'success' });

    beforeEach(() => {
      mockUseResourceQuery.mockImplementation(() => [response, refetch]);
    });

    it('should display the contents', () => {
      const { getByText } = renderIndexPage({});

      const contents = getByText('Missing <Table /> Component');
      expect(contents).toBeVisible();
    });

    it('should match the snapshot', () => {
      const { asFragment } = renderIndexPage({});

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with Table: value', () => {
    it('should display the contents', () => {
      const { getByText } = renderIndexPage({ Table });

      const contents = getByText('There are 0 books!');
      expect(contents).toBeVisible();
    });

    it('should match the snapshot', () => {
      const { asFragment } = renderIndexPage({ Table });

      expect(asFragment()).toMatchSnapshot();
    });

    describe('when the query returns a failing response', () => {
      const response = responseWithStatus({ status: 'failure' });

      beforeEach(() => {
        mockUseResourceQuery.mockImplementation(() => [response, refetch]);
      });

      it('should display the contents', () => {
        const { getByText } = renderIndexPage({ Table });

        const contents = getByText('There are 0 books!');
        expect(contents).toBeVisible();
      });

      it('should match the snapshot', () => {
        const { asFragment } = renderIndexPage({ Table });

        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('when the query returns a loading response', () => {
      const response = responseWithStatus({ status: 'loading' });

      beforeEach(() => {
        mockUseResourceQuery.mockImplementation(() => [response, refetch]);
      });

      it('should display the loading message', () => {
        const { getByText } = renderIndexPage({});

        const contents = getByText('Loading Rare Books...');
        expect(contents).toBeVisible();
      });

      it('should match the snapshot', () => {
        const { asFragment } = renderIndexPage({ Table });

        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('when the query returns a successful response', () => {
      const response = responseWithStatus({ status: 'success' });

      beforeEach(() => {
        mockUseResourceQuery.mockImplementation(() => [response, refetch]);
      });

      it('should display the contents', () => {
        const { getByText } = renderIndexPage({ Table });

        const contents = getByText('There are 0 books!');
        expect(contents).toBeVisible();
      });

      it('should match the snapshot', () => {
        const { asFragment } = renderIndexPage({ Table });

        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('when the query returns a successful response with data', () => {
      const data = {
        rareBooks: [
          { title: 'Gideon the Ninth' },
          { title: 'Harrow the Ninth' },
          { title: 'Nona the Ninth' },
        ],
      };
      const response = responseWithData({ data });

      beforeEach(() => {
        mockUseResourceQuery.mockImplementation(() => [response, refetch]);
      });

      it('should display the contents', () => {
        const { getByText } = renderIndexPage({ Table });

        const contents = getByText('There are 3 books!');
        expect(contents).toBeVisible();
      });

      it('should match the snapshot', () => {
        const { asFragment } = renderIndexPage({ Table });

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
