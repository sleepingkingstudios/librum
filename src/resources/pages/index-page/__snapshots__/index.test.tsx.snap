// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResourceIndexPage /> should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Rare Books
    </h1>
    <div
      class="min-h-[10rem] relative resources-table"
    >
      <div
        class="min-h-[10rem] relative resources-table"
      >
        <h3
          class="my-5 text-danger"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-skull-crossbones "
            data-icon="skull-crossbones"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M368 128c0 44.4-25.4 83.5-64 106.4V256c0 17.7-14.3 32-32 32H176c-17.7 0-32-14.3-32-32V234.4c-38.6-23-64-62.1-64-106.4C80 57.3 144.5 0 224 0s144 57.3 144 128zM168 176a32 32 0 1 0 0-64 32 32 0 1 0 0 64zm144-32a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zM3.4 273.7c7.9-15.8 27.1-22.2 42.9-14.3L224 348.2l177.7-88.8c15.8-7.9 35-1.5 42.9 14.3s1.5 35-14.3 42.9L295.6 384l134.8 67.4c15.8 7.9 22.2 27.1 14.3 42.9s-27.1 22.2-42.9 14.3L224 419.8 46.3 508.6c-15.8 7.9-35 1.5-42.9-14.3s-1.5-35 14.3-42.9L152.4 384 17.7 316.6C1.9 308.7-4.5 289.5 3.4 273.7z"
              fill="currentColor"
            />
          </svg>
          <span
            class="px-2"
          >
            Missing &lt;Table /&gt; Component
          </span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-skull-crossbones "
            data-icon="skull-crossbones"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M368 128c0 44.4-25.4 83.5-64 106.4V256c0 17.7-14.3 32-32 32H176c-17.7 0-32-14.3-32-32V234.4c-38.6-23-64-62.1-64-106.4C80 57.3 144.5 0 224 0s144 57.3 144 128zM168 176a32 32 0 1 0 0-64 32 32 0 1 0 0 64zm144-32a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zM3.4 273.7c7.9-15.8 27.1-22.2 42.9-14.3L224 348.2l177.7-88.8c15.8-7.9 35-1.5 42.9 14.3s1.5 35-14.3 42.9L295.6 384l134.8 67.4c15.8 7.9 22.2 27.1 14.3 42.9s-27.1 22.2-42.9 14.3L224 419.8 46.3 508.6c-15.8 7.9-35 1.5-42.9-14.3s-1.5-35 14.3-42.9L152.4 384 17.7 316.6C1.9 308.7-4.5 289.5 3.4 273.7z"
              fill="currentColor"
            />
          </svg>
        </h3>
      </div>
    </div>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourceIndexPage /> when the query returns a failing response should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Rare Books
    </h1>
    <div
      class="min-h-[10rem] relative resources-table"
    >
      <div
        class="min-h-[10rem] relative resources-table"
      >
        <h3
          class="my-5 text-danger"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-skull-crossbones "
            data-icon="skull-crossbones"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M368 128c0 44.4-25.4 83.5-64 106.4V256c0 17.7-14.3 32-32 32H176c-17.7 0-32-14.3-32-32V234.4c-38.6-23-64-62.1-64-106.4C80 57.3 144.5 0 224 0s144 57.3 144 128zM168 176a32 32 0 1 0 0-64 32 32 0 1 0 0 64zm144-32a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zM3.4 273.7c7.9-15.8 27.1-22.2 42.9-14.3L224 348.2l177.7-88.8c15.8-7.9 35-1.5 42.9 14.3s1.5 35-14.3 42.9L295.6 384l134.8 67.4c15.8 7.9 22.2 27.1 14.3 42.9s-27.1 22.2-42.9 14.3L224 419.8 46.3 508.6c-15.8 7.9-35 1.5-42.9-14.3s-1.5-35 14.3-42.9L152.4 384 17.7 316.6C1.9 308.7-4.5 289.5 3.4 273.7z"
              fill="currentColor"
            />
          </svg>
          <span
            class="px-2"
          >
            Missing &lt;Table /&gt; Component
          </span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-skull-crossbones "
            data-icon="skull-crossbones"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M368 128c0 44.4-25.4 83.5-64 106.4V256c0 17.7-14.3 32-32 32H176c-17.7 0-32-14.3-32-32V234.4c-38.6-23-64-62.1-64-106.4C80 57.3 144.5 0 224 0s144 57.3 144 128zM168 176a32 32 0 1 0 0-64 32 32 0 1 0 0 64zm144-32a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zM3.4 273.7c7.9-15.8 27.1-22.2 42.9-14.3L224 348.2l177.7-88.8c15.8-7.9 35-1.5 42.9 14.3s1.5 35-14.3 42.9L295.6 384l134.8 67.4c15.8 7.9 22.2 27.1 14.3 42.9s-27.1 22.2-42.9 14.3L224 419.8 46.3 508.6c-15.8 7.9-35 1.5-42.9-14.3s-1.5-35 14.3-42.9L152.4 384 17.7 316.6C1.9 308.7-4.5 289.5 3.4 273.7z"
              fill="currentColor"
            />
          </svg>
        </h3>
      </div>
    </div>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourceIndexPage /> when the query returns a loading response should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Rare Books
    </h1>
    <div
      class="min-h-[10rem] relative resources-table"
    >
      <div
        class="form-loading-overlay absolute grid h-full place-content-center w-full"
      >
        <div
          class="inline-block text-center"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-gear animate-spin motion-reduce:animate-none h-10 w-10"
            data-icon="gear"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M481.9 166.6c3.2 8.7 .5 18.4-6.4 24.6l-30.9 28.1c-7.7 7.1-11.4 17.5-10.9 27.9c.1 2.9 .2 5.8 .2 8.8s-.1 5.9-.2 8.8c-.5 10.5 3.1 20.9 10.9 27.9l30.9 28.1c6.9 6.2 9.6 15.9 6.4 24.6c-4.4 11.9-9.7 23.3-15.8 34.3l-4.7 8.1c-6.6 11-14 21.4-22.1 31.2c-5.9 7.2-15.7 9.6-24.5 6.8l-39.7-12.6c-10-3.2-20.8-1.1-29.7 4.6c-4.9 3.1-9.9 6.1-15.1 8.7c-9.3 4.8-16.5 13.2-18.8 23.4l-8.9 40.7c-2 9.1-9 16.3-18.2 17.8c-13.8 2.3-28 3.5-42.5 3.5s-28.7-1.2-42.5-3.5c-9.2-1.5-16.2-8.7-18.2-17.8l-8.9-40.7c-2.2-10.2-9.5-18.6-18.8-23.4c-5.2-2.7-10.2-5.6-15.1-8.7c-8.8-5.7-19.7-7.8-29.7-4.6L69.1 425.9c-8.8 2.8-18.6 .3-24.5-6.8c-8.1-9.8-15.5-20.2-22.1-31.2l-4.7-8.1c-6.1-11-11.4-22.4-15.8-34.3c-3.2-8.7-.5-18.4 6.4-24.6l30.9-28.1c7.7-7.1 11.4-17.5 10.9-27.9c-.1-2.9-.2-5.8-.2-8.8s.1-5.9 .2-8.8c.5-10.5-3.1-20.9-10.9-27.9L8.4 191.2c-6.9-6.2-9.6-15.9-6.4-24.6c4.4-11.9 9.7-23.3 15.8-34.3l4.7-8.1c6.6-11 14-21.4 22.1-31.2c5.9-7.2 15.7-9.6 24.5-6.8l39.7 12.6c10 3.2 20.8 1.1 29.7-4.6c4.9-3.1 9.9-6.1 15.1-8.7c9.3-4.8 16.5-13.2 18.8-23.4l8.9-40.7c2-9.1 9-16.3 18.2-17.8C213.3 1.2 227.5 0 242 0s28.7 1.2 42.5 3.5c9.2 1.5 16.2 8.7 18.2 17.8l8.9 40.7c2.2 10.2 9.4 18.6 18.8 23.4c5.2 2.7 10.2 5.6 15.1 8.7c8.8 5.7 19.7 7.7 29.7 4.6l39.7-12.6c8.8-2.8 18.6-.3 24.5 6.8c8.1 9.8 15.5 20.2 22.1 31.2l4.7 8.1c6.1 11 11.4 22.4 15.8 34.3zM242 336a80 80 0 1 0 0-160 80 80 0 1 0 0 160z"
              fill="currentColor"
            />
          </svg>
          <br />
          Loading Rare Books...
        </div>
      </div>
    </div>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourceIndexPage /> when the query returns a successful response should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Rare Books
    </h1>
    <div
      class="min-h-[10rem] relative resources-table"
    >
      <div
        class="min-h-[10rem] relative resources-table"
      >
        <h3
          class="my-5 text-danger"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-skull-crossbones "
            data-icon="skull-crossbones"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M368 128c0 44.4-25.4 83.5-64 106.4V256c0 17.7-14.3 32-32 32H176c-17.7 0-32-14.3-32-32V234.4c-38.6-23-64-62.1-64-106.4C80 57.3 144.5 0 224 0s144 57.3 144 128zM168 176a32 32 0 1 0 0-64 32 32 0 1 0 0 64zm144-32a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zM3.4 273.7c7.9-15.8 27.1-22.2 42.9-14.3L224 348.2l177.7-88.8c15.8-7.9 35-1.5 42.9 14.3s1.5 35-14.3 42.9L295.6 384l134.8 67.4c15.8 7.9 22.2 27.1 14.3 42.9s-27.1 22.2-42.9 14.3L224 419.8 46.3 508.6c-15.8 7.9-35 1.5-42.9-14.3s-1.5-35 14.3-42.9L152.4 384 17.7 316.6C1.9 308.7-4.5 289.5 3.4 273.7z"
              fill="currentColor"
            />
          </svg>
          <span
            class="px-2"
          >
            Missing &lt;Table /&gt; Component
          </span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-skull-crossbones "
            data-icon="skull-crossbones"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M368 128c0 44.4-25.4 83.5-64 106.4V256c0 17.7-14.3 32-32 32H176c-17.7 0-32-14.3-32-32V234.4c-38.6-23-64-62.1-64-106.4C80 57.3 144.5 0 224 0s144 57.3 144 128zM168 176a32 32 0 1 0 0-64 32 32 0 1 0 0 64zm144-32a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zM3.4 273.7c7.9-15.8 27.1-22.2 42.9-14.3L224 348.2l177.7-88.8c15.8-7.9 35-1.5 42.9 14.3s1.5 35-14.3 42.9L295.6 384l134.8 67.4c15.8 7.9 22.2 27.1 14.3 42.9s-27.1 22.2-42.9 14.3L224 419.8 46.3 508.6c-15.8 7.9-35 1.5-42.9-14.3s-1.5-35 14.3-42.9L152.4 384 17.7 316.6C1.9 308.7-4.5 289.5 3.4 273.7z"
              fill="currentColor"
            />
          </svg>
        </h3>
      </div>
    </div>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourceIndexPage /> with Table: value should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Rare Books
    </h1>
    <div
      class="min-h-[10rem] relative resources-table"
    >
      <span
        class="mock-data-table"
      >
        There are 0 books!
      </span>
    </div>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourceIndexPage /> with Table: value when the query returns a failing response should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Rare Books
    </h1>
    <div
      class="min-h-[10rem] relative resources-table"
    >
      <span
        class="mock-data-table"
      >
        There are 0 books!
      </span>
    </div>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourceIndexPage /> with Table: value when the query returns a loading response should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Rare Books
    </h1>
    <div
      class="min-h-[10rem] relative resources-table"
    >
      <div
        class="form-loading-overlay absolute grid h-full place-content-center w-full"
      >
        <div
          class="inline-block text-center"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-gear animate-spin motion-reduce:animate-none h-10 w-10"
            data-icon="gear"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M481.9 166.6c3.2 8.7 .5 18.4-6.4 24.6l-30.9 28.1c-7.7 7.1-11.4 17.5-10.9 27.9c.1 2.9 .2 5.8 .2 8.8s-.1 5.9-.2 8.8c-.5 10.5 3.1 20.9 10.9 27.9l30.9 28.1c6.9 6.2 9.6 15.9 6.4 24.6c-4.4 11.9-9.7 23.3-15.8 34.3l-4.7 8.1c-6.6 11-14 21.4-22.1 31.2c-5.9 7.2-15.7 9.6-24.5 6.8l-39.7-12.6c-10-3.2-20.8-1.1-29.7 4.6c-4.9 3.1-9.9 6.1-15.1 8.7c-9.3 4.8-16.5 13.2-18.8 23.4l-8.9 40.7c-2 9.1-9 16.3-18.2 17.8c-13.8 2.3-28 3.5-42.5 3.5s-28.7-1.2-42.5-3.5c-9.2-1.5-16.2-8.7-18.2-17.8l-8.9-40.7c-2.2-10.2-9.5-18.6-18.8-23.4c-5.2-2.7-10.2-5.6-15.1-8.7c-8.8-5.7-19.7-7.8-29.7-4.6L69.1 425.9c-8.8 2.8-18.6 .3-24.5-6.8c-8.1-9.8-15.5-20.2-22.1-31.2l-4.7-8.1c-6.1-11-11.4-22.4-15.8-34.3c-3.2-8.7-.5-18.4 6.4-24.6l30.9-28.1c7.7-7.1 11.4-17.5 10.9-27.9c-.1-2.9-.2-5.8-.2-8.8s.1-5.9 .2-8.8c.5-10.5-3.1-20.9-10.9-27.9L8.4 191.2c-6.9-6.2-9.6-15.9-6.4-24.6c4.4-11.9 9.7-23.3 15.8-34.3l4.7-8.1c6.6-11 14-21.4 22.1-31.2c5.9-7.2 15.7-9.6 24.5-6.8l39.7 12.6c10 3.2 20.8 1.1 29.7-4.6c4.9-3.1 9.9-6.1 15.1-8.7c9.3-4.8 16.5-13.2 18.8-23.4l8.9-40.7c2-9.1 9-16.3 18.2-17.8C213.3 1.2 227.5 0 242 0s28.7 1.2 42.5 3.5c9.2 1.5 16.2 8.7 18.2 17.8l8.9 40.7c2.2 10.2 9.4 18.6 18.8 23.4c5.2 2.7 10.2 5.6 15.1 8.7c8.8 5.7 19.7 7.7 29.7 4.6l39.7-12.6c8.8-2.8 18.6-.3 24.5 6.8c8.1 9.8 15.5 20.2 22.1 31.2l4.7 8.1c6.1 11 11.4 22.4 15.8 34.3zM242 336a80 80 0 1 0 0-160 80 80 0 1 0 0 160z"
              fill="currentColor"
            />
          </svg>
          <br />
          Loading Rare Books...
        </div>
      </div>
    </div>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourceIndexPage /> with Table: value when the query returns a successful response should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Rare Books
    </h1>
    <div
      class="min-h-[10rem] relative resources-table"
    >
      <span
        class="mock-data-table"
      >
        There are 0 books!
      </span>
    </div>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourceIndexPage /> with Table: value when the query returns a successful response with data should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Rare Books
    </h1>
    <div
      class="min-h-[10rem] relative resources-table"
    >
      <span
        class="mock-data-table"
      >
        There are 3 books!
      </span>
    </div>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
